File: towers.rb

def printMove(fr, to)
  puts "move disk at peg #{fr} to peg #{to}."
end

def towers(n, fr, to, spare)  
  if n == 1
    printMove(fr, to)
  else
    towers(n-1, fr, spare, to)       
    towers(1, fr, to, spare)    
    towers(n-1, spare, to, fr)    
  end
end

towers(2, "A", "B", "C")

File: towers.py
def printMove(fr, to):
    global numCalls
    numCalls += 1
    print('move disk at peg ' + str(fr) + ' to peg ' + str(to))

def Towers(n, fr, to, spare):
    if n == 1:        
        printMove(fr, to)        
    else:        
        Towers(n-1, fr, spare, to)        
        Towers(1, fr, to, spare)        
        Towers(n-1, spare, to, fr)
        

numCalls = 0
Towers(1, 'A', 'B', 'C')
print numCalls
print
Towers(2, 'A', 'B', 'C')
print numCalls
print
Towers(3, 'A', 'B', 'C')
print numCalls
